src/apps/cbr/CbrSender.cc

```diff
@@ -110,15 +118,14 @@ void CbrSender::initTraffic()
 
 void CbrSender::sendCbrPacket()
 {
-    EV << "CbrSender::sendCbrPacket - Sending frame[" << iDframe_ << "/" << nframes_ << "], next packet at "<< simTime() + sampling_time << endl;
-
-    emit(cbrSentPktSignal_, (long)iDframe_);
-
-    CbrPacket* packet = new CbrPacket("Cbr");
-    packet->setNframes(nframes_);
-    packet->setIDframe(iDframe_++);
-    packet->setTimestamp(simTime());
-    packet->setByteLength(size_);
+    Packet* packet = new Packet("CBR");
+    auto cbr = makeShared<CbrPacket>();
+    cbr->setNframes(nframes_);
+    cbr->setIDframe(iDframe_);
+    cbr->setTimestamp(simTime());
+    cbr->setSize(size_);
```

Comment:
If setSize sets the content of a size field here (i.e. the bits in the packet), then you may consider renaming the field to sizeField (to have a setSizeField). This change makes it easier to differentiate between the length/size field and the actual length/size of the chunk.

Reply:
Good point. I guess the same is true for the timestamp field. Therefore, we renamed the "size" field within all the application layer packets to "payloadSize" and "timestamp" to "payloadTimestamp". This should make the difference more distinguishable.